scout/server/blueprints/institutes/controllers.py

```diff
@@ -514,3 +517,259 @@ def hgvs_str(gene_symbols, hgvs_p, hgvs_c):
+    """Add an empty subpanel to a phenotype model
+
+    Args:
+        model_id(str): string of the ID of a phenotype model
+        model_obj(dict): a dictionary coresponding to a phenotype model
+        user_form(request.form): a POST request form object
+
+    """
+    subpanel_key = generate_md5_key([model_id, user_form.get("title")])
+    phenomodel_subpanels = model_obj.get("subpanels") or {}
+
+    if subpanel_key in phenomodel_subpanels:
+        flash("A model panel with that title already exists", "warning")
+        return
+    subpanel_obj = {
+        "title": user_form.get("title"),
+        "subtitle": user_form.get("subtitle"),
+        "created": datetime.datetime.now(),
+        "updated": datetime.datetime.now(),
+    }
+    phenomodel_subpanels[subpanel_key] = subpanel_obj
+    model_obj["subpanels"] = phenomodel_subpanels
+    return model_obj
+
+
+def edit_subpanel_checkbox(model_id, user_form):
+    """Update checkboxes from one or more panels according to the user form
+    Args:
+        model_id(ObjectId): document ID of the model to be updated
+        user_form(request.form): a POST request form object
+    """
+    model_obj = store.phenomodel(model_id)
+    if model_obj is None:
+        return
+    if "add_hpo" in user_form:  # add an HPO checkbox to subpanel
+        model_obj = _subpanel_hpo_checkgroup_add(model_obj, user_form)
+    if "add_omim" in user_form:  # add an OMIM checkbox to subpanel
+        model_obj = _subpanel_omim_checkbox_add(model_obj, user_form)
+    if user_form.get("checkgroup_remove"):  # remove a checkbox of any type from subpanel
+        model_obj = _subpanel_checkgroup_remove_one(model_obj, user_form)
+
+    if model_obj:
+        store.update_phenomodel(model_id=model_id, model_obj=model_obj)
+
+
+def update_phenomodel(model_id, user_form):
+    """Update a phenotype model according to the user form
+
+    Args:
+        model_id(ObjectId): document ID of the model to be updated
+        user_form(request.form): a POST request form object
+    """
+    model_obj = store.phenomodel(model_id)
+    if model_obj is None:
+        return
+    if user_form.get("update_model"):  # update either model name of description
+        model_obj["name"] = user_form.get("model_name")
+        model_obj["description"] = user_form.get("model_desc")
+    if user_form.get("subpanel_delete"):  # Remove a phenotype submodel from phenomodel
+        subpanels = model_obj["subpanels"]
+        # remove panel from subpanels dictionary
+        subpanels.pop(user_form.get("subpanel_delete"), None)
+        model_obj["subpanels"] = subpanels
+    if user_form.get("add_subpanel"):  # Add a new phenotype submodel
+        model_obj = _add_subpanel(model_id, model_obj, user_form)
+    if user_form.get("model_save"):  # Save model according user preferences in the preview
+        model_obj = phenomodel_checkgroups_filter(model_obj, user_form)
+
+    if model_obj:
```

Comment:
Same here: consider updated_model_obj or some other way to indicated that this is not supposed to be the same as before. Seems a bit unnecessary to update if nothing actually changed e.g.?

Reply:
Fair enough: I've created a boolean variable named update_dict that is False then becomes True as model_obj (model_dict) changes

Reply:
No, I'm naming it update_model actually.